code/game/machinery/turnstile.dm

```diff
@@ -0,0 +1,66 @@
+/obj/machinery/turnstile
+	name = "turnstile"
+	desc = "A remote control switch."
+	icon = 'icons/obj/turnstile.dmi'
+	icon_state = "turnstile_map"
+	power_channel = ENVIRON
+	density = TRUE
+	armor = list(melee = 50, bullet = 50, laser = 50, energy = 50, bomb = 10, bio = 100, rad = 100, fire = 90, acid = 70)
+	anchored = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 2
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	layer = OPEN_DOOR_LAYER
+
+/obj/machinery/turnstile/Initialize()
+	. = ..()
+	icon_state = "turnstile"
+
+/obj/machinery/turnstile/CollidedWith(atom/movable/AM)
+	if(ismob(AM))
+		var/mob/B = AM
+		if(isliving(AM))
+			var/mob/living/M = AM
+
+			if(world.time - M.last_bumped <= 10)
+				return
+			M.last_bumped = world.time
+
+			var/allowed_access = FALSE
+			var/turf/behind = get_step(src, dir)
+
+			if(AM in behind.contents)
+				allowed_access = allowed(B)
+			else
+				to_chat(usr, "<span class='notice'>\the [src] resists your efforts.</span>")
+				return
+
+			if(allowed_access)
+				var/mob/living/carbon/human/HU = AM
+				if(istype(HU))
+					var/obj/item/restraints/handcuffs/cable/zipties = HU.handcuffed
+					if(istype(zipties))
+						qdel(zipties)
+						to_chat(usr, "<span class='notice'>\the [src] cuts off \the [zipties].</span>")

```

Comment:
It really shouldn't cut off zipties.

Two prisoners are fighting, security gets in, cuffs both and tries to exit. Bam, prisoner is free because they didn't know about a line of code.

Reply:
If everyones cuffed, then they can exit via the door, not the turnstile.

The reason this cuts zipties is because if you push them into genpop, what is to stop the person already in genpop to drag them around forever, never uncuffing them?